### PR TITLE
Cylindrical billboard implementation

### PR DESCRIPTION
**Pull request by tvcastillod**
The current billboard actor only supports spherical billboards functionality, this PR is intended to add the option of using cylindrical billboards.

### CODE REVIEW

**Review by guaje — CHANGES_REQUESTED** [review]
@tvcastillod, thanks for this PR. Overall great job. Here are some comments:

File: fury/tests/test_actors.py [hunk 1]
```diff
@@ -1389,7 +1389,7 @@ def test_superquadric_actor(interactive=False):
     npt.assert_equal(report.colors_found, [True, True, True])
 
 
-def test_billboard_actor(interactive=False):
+def test_billboard_actor(interactive=True):
```
Comment (guaje) [inline fury/tests/test_actors.py:1392]: Change interactive to `false` so it doesn't freeze the main thread.

**Review by guaje — CHANGES_REQUESTED** [review]
Besides the only comment on this review, I think this PR is ready to be merged.

**Comment by tvcastillod** [comment]
Hi @Garyfallidis, may you review this PR?

File: fury/tests/test_actors.py [hunk 2]
```diff
@@ -1560,13 +1559,121 @@ def test_billboard_actor(interactive=False):
...
+    scene.yaw(87)
+    if interactive:
+        window.show(scene)
+
+    scene.reset_camera()
+    scene.reset_clipping_range()
+    arr = window.snapshot(scene, offscreen=True)
+    report = window.analyze_snapshot(arr, colors=255 * colors)
+    npt.assert_equal(report.colors_found, [True] * 5)
+
```
Comment (skoudoro) [inline fury/tests/test_actors.py:1616]: Huge test for doing the exact same things.   Can you just create a small loop? thanks

File: fury/actor.py [hunk 3]
```diff
@@ -2699,6 +2701,11 @@ def billboard(
     fs_impl : str or list of str, optional
         Fragment Shaders code that contains all variable/function
         implementation.
+    bb_type : str
```
Comment (skoudoro) [inline fury/actor.py:2704]: `str, optional` instead

**Review by skoudoro — COMMENTED** [review]
Hi @tvcastillod,

Thank for this. Here, an initial review. I will go deeper a bit later:

- Concerning the shader file extension. it would be great if it is "*.frag" instead of "*.glsl"
- Also, it helps a lot if the filename ends by `_impl` or `_dec` to know if we just declare new function or update the core functionality
- It would be great to have a tutorial or update the billboard tutorial if it exists. Otherwise, it will not be used.

File: fury/actor.py [hunk 4]
```diff
@@ -2720,12 +2727,61 @@ def billboard(
...
+         compose_shader(bb_norm) +
+         compose_shader(bb_type_sd)])
     vs_impl_code = compose_shader(
-        [compose_shader(vs_impl) + import_fury_shader('billboard_impl.vert')]
-    )
+        [compose_shader(vs_impl) + billboard_vert_impl])
+
     gs_code = compose_shader(gs_prog)
     fs_dec_code = compose_shader(
         [import_fury_shader('billboard_dec.frag') + compose_shader(fs_dec)]
```
Comment (guaje) [inline fury/actor.py:2787]: Please add the content of `billboard_dec.frag`  to the python implementation and remove the file to comply with #753. Additionally, replace `+` with `,` when using the `compose_shader` function.

**Review by guaje — CHANGES_REQUESTED** [review]
@tvcastillod, thanks for adding the requested changes. After a maintainers meeting, we agreed on changes related to adding shaders to Fury. The summary can be found in #753, and therefore the comments in this review are related to that.

**Review by skoudoro — COMMENTED** [review]
Hi @tvcastillod,

I suppose you made a mistake during your last commit named "solve conflict".  You merged the wrong version which create a mess. I recommend you to rebase this PR.

Currently, you are reverting many fixes.

Let us know if you need help

**Comment by guaje** [comment]
@tvcastillod, thanks for rebasing and amending the PR to comply with #753. I don't have any more comments. @skoudoro what do you think?

**Review by skoudoro — APPROVED** [review]
All Good, Thank you for this work @tvcastillod, merging!

Now, Looking forward for an example of use!!

**Pull request merged**